Zewa/Router.php

```diff
@@ -141,9 +141,9 @@ private function normalize($data)
         if (is_numeric($data)) {
             if (is_int($data) || ctype_digit(trim($data, '-'))) {
                 $data = (int)$data;
-            } else if ($data == (string)(float)$data) {
+            } else if ($data === (string)(float)$data) {
```

Comment:
This is the actual fix...   This PR would be super tiny with a couple lines changed if it wasn't for all the crap I had to add to implement testing.